### PR TITLE
Added description to widget example programs except Cursor and Menu

### PR DESCRIPTION
**Pull request by ash13**
## PR Summary

Was working on  #8921 to add one line description for the example programs: I worked on the example programs related to widgets. I was not able to check the Curosr and the menu widget. Other than those widgets, I have added description to the rest. Please do review and let me know if it needs any changes. 

examples/widgets/buttons.py - Done 
examples/widgets/check_buttons.py - Done 
examples/widgets/cursor.py - Had few errors in my system. Will work on it. Not done.
examples/widgets/lasso_selector_demo_sgskip.py - Done
examples/widgets/menu.py - Din't work on my system. Will work on it.
examples/widgets/multicursor.py - Done
examples/widgets/radio_buttons.py - Done
examples/widgets/slider_demo.py- Done
examples/widgets/textbox.py - Done

LINK: https://github.com/matplotlib/matplotlib/issues/8921

## PR Checklist

- [ ] Has Pytest style unit tests
- [ ] Code is PEP 8 compliant 
- [ ] New features are documented, with examples if plot related
- [ ] Documentation is sphinx and numpydoc compliant
- [ ] Added an entry to doc/users/next_whats_new/ if major new feature (follow instructions in README.rst there)
- [ ] Documented in doc/api/api_changes.rst if API changed in a backward-incompatible way

### CODE REVIEW

File: examples/widgets/buttons.py [hunk 1]
```diff
@@ -2,7 +2,7 @@
 =======
 Buttons
 =======
-
+An pretty cool example to show how to use buttons widget.The program shows a simple GUI with a sine wave changing its frequency.The next and previous button widget used here helps shows the wave with new frequency.
```
Comment (divyam3897) [inline examples/widgets/buttons.py]: An pretty cool? You should not have had the first line, a small description should be there. Don't use the words like "an example to show etc"

File: examples/widgets/slider_demo1.py [hunk 2]
```diff
@@ -0,0 +1,55 @@
+"""
+===========
+Slider Demo
+===========
+
```
Comment (divyam3897) [inline examples/widgets/slider_demo1.py]: There is no description added, why changes in the whole file?

File: examples/widgets/buttons.py (hunk 1 above)
Comment (ash13) [reply examples/widgets/buttons.py]: Sorry it was 'A' . And yes will remove the first line.

File: examples/widgets/slider_demo1.py (hunk 2 above)
Comment (ash13) [reply examples/widgets/slider_demo1.py]: Sorry about that. I think I din't save the changes. I will fix it.

**Comment by ash13** [comment]
Fixed it. Could you please check.

**Comment by NelleV** [comment]
Hi @ash13 
We are following the pep8 conventions in our code. The lines should be max 79 char long, so the description you've added are too long. Do you mind breaking the lines?

**Comment by NelleV** [comment]
Thanks @ash13 and @choldgraf !

**Pull request merged**